jiant/tasks/lm.py

```diff
@@ -246,16 +246,26 @@ def update_metrics(self, out, batch=None):
         return
 
     def get_data_iter(self, path):
-        """Loading data file and tokenizing the text
+        """
+        Loading data file and tokenizing the text. We treat the Wikipedia corpus as a
+        long sequence, and we take each slice of 510 tokens as an example. The dataset
+        consists of a sentence per row in the file. This function concatenates all of the sentences,
+        before going through the sequence and yielding each chunk of 510 tokens.
         Args:
             path: (str) data file path
         """
+        seq_len = self.max_seq_len - 2
+        total_tokens = []
         with open(path, "r", encoding="utf-8") as txt_fh:
             for row in txt_fh:
                 toks = row.strip()
                 if not toks:
                     continue
-                yield tokenize_and_truncate(self._tokenizer_name, toks, self.max_seq_len)
+                toks = tokenize_and_truncate(self._tokenizer_name, toks, -1)
```

Comment:
Why drop the last 3 tokens from every line?

Reply:
Good point, that's not supposed to happen.